SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestHandover/SpatialTestHandover.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestHandover.h"
+#include "Kismet/GameplayStatics.h"
+#include "HandoverCube.h"
+#include "SpatialFunctionalTestFlowController.h"
+
+/**
+ * This test automated the Handover Gym which demonstrates that Entities correctly migrate between areas of authority.
+ * This test requires the map it runs in to have the Multi worker enabled, with the BP_QuadrantZoningSettings set in order to be relevant.
+ * The test includes 4 servers and 2 client workers.
+ * 
+ * The flow is as follows:
+ * - Setup:
+ *	- Server 1 spawns a HandoverCube inside its authority area.
+ *  - All servers set a reference to the HandoverCube and reset their local copy of the LocationIndex and AuthorityCheckIndex.
+ * - Test:
+ *	- At this stage, Server 1 should have authority over the HandoverCube.
+ *  - The HandoverCube is moved in the authority area of Server 2.
+ *  - At this stage, Server 2 should have authority over the HandoverCube.
+ *  - Server 2 acquires a lock on the HandoverCube and moves it into the authority area of Server 4.
+ *	- Since Server 2 has the lock on the HandoverCube it should still be authoritative over it.
+ *  - Server 2 releases the lock on the HandoverCube.
+ *  - At this point, Server 4 should become authoritative over the HandoverCube.
+ *  - The HandoverCube is moved in the authority area of Server 3.
+ *  - At this point, Server 3 should be authoritative over the HandoverCube.
+ * - Clean-up:
+ *	- The HandoverCube is destroyed.
+ */
+ASpatialTestHandover::ASpatialTestHandover()
+	: Super()
+{
+	Author = "Andrei";
+	Description = TEXT("Test Actor handover");
+
+	SpawnLocation = FVector(-500.0f, -500.0f, 50.0f);
+
+	TestLocations.Add(FVector(500.0f, -500.0f, 50.0f));
+	TestLocations.Add(FVector(500.0f, 500.0f, 50.0f)); 
+	TestLocations.Add(FVector(-500.0f, 500.0f, 50.0f));
+
+	ExpectedAuthoritativeServer.Add(1);
```

Comment:
This relies on knowing in advance how the LBS will allocate the virtual worker Ids, and may be brittle. You could instead interrogate the NetDriver's LoadBalancingStrategy, like I do in https://github.com/spatialos/UnrealGDK/pull/2447 -> SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DebugInterface/TestDebugInterface.cpp

This is not ideal either, and assumes you know which LBS is set on the map, but is not tied to the area allocation scheme anymore.

Reply:
Yes that is correct, I will take a look and modify this accordingly, thank you for the suggestion!